core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -972,14 +956,16 @@ public GHResponse route(GHRequest request) {
             if (!encodingManager.hasEncoder(vehicle))
                 throw new IllegalArgumentException("Vehicle not supported: " + vehicle + ". Supported are: " + encodingManager.toString());
 
+            FlagEncoder encoder = encodingManager.getEncoder(vehicle);
             HintsMap hints = request.getHints();
-            String tModeStr = hints.get("traversal_mode", traversalMode.toString());
-            TraversalMode tMode = TraversalMode.fromString(tModeStr);
+
+            TraversalMode tMode = encoder.supports(TurnWeighting.class) ? TraversalMode.EDGE_BASED : TraversalMode.NODE_BASED;
+
+            String tModeStr = hints.get("traversal_mode", tMode.toString());
```

Comment:
Can I remove this `traversal_mode` parameter to avoid further confusion ? Right now it almost does the same as the `edge_based` parameter and I did not find documentation for it anywhere ?

Reply:
Yes, let's do this.